File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/DestructReferenceHandler.py
from javonet.core.referenceCache.ReferencesCache import ReferencesCache

from javonet.core.handler.AbstractCommandHandler import *


class DestructReferenceHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 1

    def process(self, command):
        reference_cache = ReferencesCache()
        return reference_cache.delete_reference(command.payload[0])

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/MacOs/X64/core/handler/destruct_reference_handler.rb
require_relative 'abstract_command_handler'
require_relative '../reference_cache/references_cache'

class DestructReferenceHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 1
  end

  def process(command)
    begin
      if command.payload.length == @required_parameters_count
        reference_cache = ReferencesCache.instance
        return reference_cache.delete_reference(command.payload[0])
      else
        raise ArgumentError.new "Destruct Reference Handler parameters mismatch"
      end
    rescue Exception => e
      return e
    end
  end
end